src/services/earn/earn-service.ts

```diff
@@ -128,8 +128,8 @@ export class EarnService implements IEarnService {
     } else if (usePermit2) {
       return PERMIT2_CONTRACT.address(chainId);
     }
-
-    const { needsSwap } = await this.checkIfNeedsSwap({ chainId, strategyId, depositToken: depositWith });
+    const [, , tokenId] = strategyId.split('-');
```

Comment:
`tokenId` is called like that on the position, because it represents the NFT (token) id. In this case, we should call it differently. Like `numericStrategyId` or `strategyIdInChain`, or other